Mobile: Improve npm clean scripts for react-native-editor

mchowning opened the pull request:
## Description
Updating the clean scripts to delete the proper cache files.

It seems like these cache files have moved (possibly due to one of the React Native upgrades) since these scripts were last updated. I am basing these updates on the files that I am seeing on my local machine. Let me know if you think that these might be different for different people.

I did notice that there was also a `kotlin-compiler-in-wordpressreactnativebridge-17156846827643056472.alive` in the `$TMPDIR`. I am not adding anything to remove that at this time since I don't know what it is used for, but it might be something we want to consider in the future.

## Testing Instructions
1. Run `npm run bundle` from gutenberg-mobile
2. Observe that there is at least one `haste-map-metro-*` file and a `metro-cache/` file in "$TMPDIR"
3. Run `npm run clean` from gutenberg-mobile
4. Observe that the files from step 2 no longer exist, and that there are no other recently added or updated files relating to react-native, metro, or haste in `$TMPDIR/` (apart from the earlier mentioned `kotlin-compiler-in...` file.


## Types of changes
Infrastructure improvement

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [ ] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [ ] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [X] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [X] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [X]  I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->
- [X] I've updated related schemas if appropriate. <!-- Reference: https://github.com/WordPress/gutenberg/tree/trunk/schemas -->

mchowning committed:
Update reference to metro cache

mchowning committed:
Update path to haste cache

mchowning commented on packages/react-native-editor/package.json:
```diff
@@ -123,10 +123,10 @@
 		"clean:jest": "jest --clearCache --config ../../test/native/jest.config.js; rm -rf $TMPDIR/jest_*",
-		"clean:metro": "rm -rf $TMPDIR/metro-cache-*;",
+		"clean:metro": "rm -rf $TMPDIR/metro-cache/",
 		"clean:react": "rm -rf $TMPDIR/react-*",
```
I'm not seeing any `$TMPDIR/react-*` files. Perhaps we don't need the `clean:react` command anymore?

jhnstn replied:
I don't have any `$TMPDIR/react-*` either, probably ok to remove the command.

mchowning replied:
👍  Updated the PR to remove `clean:react` as well. I'll re-request your review in case you have any additional thoughts.

jhnstn reviewed (approved):
`npm run clean` worked as expected, 👍

mchowning committed:
Remove clean:react command

These files don't seem to get created anymore, so we don't need to
clean them either

jhnstn reviewed (approved):
👍
